docs/msbuild/msbuild-conditions.md

```diff
@@ -60,6 +60,15 @@ Boolean logic is only evaluated in the context of conditions, so property settin
 
 MSBuild implements a few special processing rules to make it easier to work with string properties that are used as Boolean values. Boolean literals are accepted, so `Condition="true"` and `Condition="false"` work as expected. MSBuild also includes special rules to support the Boolean negation operator. So, if `$(Prop)` is 'true', `!$(Prop)` expands to `!true` and this compares equal to `false`, as you would expect.
 
+## Comparing versions
+
+The relational operators `<`, `>`, `<=`, and `>=` support versions as parsed by <xref:System.Version?displayProperty=fullName>, so you can compare versions that have four numeric parts to each other. For instance `'1.2.3.4' < '1.10.0.0'` is `true`.
+
+> [!CAUTION]
+> `System.Version` comparisons can produce surprising results when one or both versions do not specify all four parts. For instance, version 1.1 is older than version 1.1.0.
+
+MSBuild provides [property functions to compare versions](property-functions.md#MSBuild-version-comparison-functions) that have a different set of rules compatible with semantic versioning (semver).
+
```

Comment:
It's clear to me, but a examples wouldn't hurt.

Reply:
I agree with you but don't have time at the moment to create them. I think it's better with this PR even with no examples than the current state.

Reply:
Borrowing from the original PR, this test code is pretty clear:

```C#
[Theory]
        [InlineData("v1.0", "2.1", -1)]
        [InlineData("3.2", "3.14-pre", -1)]
        [InlineData("3+metadata", "3.0", 0)]
        [InlineData("2.1", "2.1.0", 0)]
        [InlineData("v1.2.3-pre+metadata", "1.2.3.0", 0)]
        [InlineData("3.14", "3.2", 1)]
        [InlineData("42.43.44.45", "42.43.44.5", 1)]
        public void PropertyFunctionVersionComparisons(string a, string b, int expectedSign)
        {
            var pg = new PropertyDictionary<ProjectPropertyInstance>();
            var expander = new Expander<ProjectPropertyInstance, ProjectItemInstance>(pg, FileSystems.Default);

            AssertSuccess(expectedSign >  0, $"$([MSBuild]::VersionGreaterThan('{a}', '{b}'))");
            AssertSuccess(expectedSign >= 0, $"$([MSBuild]::VersionGreaterThanOrEquals('{a}', '{b}'))");
            AssertSuccess(expectedSign <  0, $"$([MSBuild]::VersionLessThan('{a}', '{b}'))");
            AssertSuccess(expectedSign <= 0, $"$([MSBuild]::VersionLessThanOrEquals('{a}', '{b}'))");
            AssertSuccess(expectedSign == 0, $"$([MSBuild]::VersionEquals('{a}', '{b}'))");
            AssertSuccess(expectedSign != 0, $"$([MSBuild]::VersionNotEquals('{a}', '{b}'))");

            void AssertSuccess(bool expected, string expression)
            {
                bool actual = (bool)expander.ExpandPropertiesLeaveTypedAndEscaped(
                    expression,
                    ExpanderOptions.ExpandProperties,
                    MockElementLocation.Instance);

                Assert.Equal(expected, actual);
            }
        }
```

Would we have to make that not-code before including it in documentation?